ffconv/stream_processors.py

```diff
@@ -0,0 +1,240 @@
+"""
+This module contains the actual stream processors, which do the actual
+media conversions.
+"""
+
+from .utils import execute_cmd, log, CalledProcessError
+
+
+class StreamProcessor(object):
+    """
+    Base class for all stream processors, providing the basic interface and
+    common functionality.
+    """
+    media_type = None
+
+    def __init__(self, input, stream, profile):
+        """
+        Set generic input and target specs from input file, stream and profile.
+        """
+        # Set direct values from input and stream
+        self.input = input
+        self.index = stream['index']
+        self.codec = stream['codec_name']
+        self.language = stream.get('tags', {}).get('language')
+        if self.language == 'und':
+            self.language = None
+        self.allowed_codecs = profile[self.media_type]['codecs']
+
+        # Select target values from profile
+        self.target_codec = self.allowed_codecs[0]
+        self.target_container = profile[self.media_type]['container']
+        self.output = '{}-{}.{}'.format(self.media_type, self.index, self.target_container)
+
+        # Set stream-specific data
+        self._init_stream(input, stream, profile)
+
+    def _init_stream(self, input, stream, profile):
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Redefining built-in 'input' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=Pylint%3AW0622)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Unused argument 'profile' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=Pylint%3AW0613)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Unused argument 'input' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=Pylint%3AW0613)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Unused argument 'stream' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=Pylint%3AW0613)